commerce-subscription-web/src/main/resources/META-INF/resources/definition/edit_definition_subscription_info.jsp

```diff
@@ -0,0 +1,203 @@
+<%--
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+--%>
+
+<%@ include file="/init.jsp" %>
+
+<%
+String lifecycle = (String)request.getAttribute(LiferayPortletRequest.LIFECYCLE_PHASE);
+
+PortletURL catalogURLObj = PortalUtil.getControlPanelPortletURL(request, CPPortletKeys.CP_DEFINITIONS, lifecycle);
+
+String catalogURL = catalogURLObj.toString();
+
+CPDefinitionSubscriptionInfoDisplayContext cpDefinitionSubscriptionInfoDisplayContext = (CPDefinitionSubscriptionInfoDisplayContext)request.getAttribute(WebKeys.PORTLET_DISPLAY_CONTEXT);
+
+CPDefinition cpDefinition = cpDefinitionSubscriptionInfoDisplayContext.getCPDefinition();
+long cpDefinitionId = cpDefinitionSubscriptionInfoDisplayContext.getCPDefinitionId();
+List<CPSubscriptionType> cpSubscriptionTypes = cpDefinitionSubscriptionInfoDisplayContext.getCPSubscriptionTypes();
+
+String defaultCPSubscriptionType = StringPool.BLANK;
+
+if (!cpSubscriptionTypes.isEmpty()) {
+	CPSubscriptionType firstCPSubscriptionType = cpSubscriptionTypes.get(0);
+
+	defaultCPSubscriptionType = firstCPSubscriptionType.getName();
+}
+
+boolean subscriptionEnabled = BeanParamUtil.getBoolean(cpDefinition, request, "subscriptionEnabled", false);
+int subscriptionLength = BeanParamUtil.getInteger(cpDefinition, request, "subscriptionLength", 1);
+String subscriptionType = BeanParamUtil.getString(cpDefinition, request, "subscriptionType", defaultCPSubscriptionType);
+long maxSubscriptionCycles = BeanParamUtil.getLong(cpDefinition, request, "maxSubscriptionCycles");
+
+String defaultCPSubscriptionTypeLabel = StringPool.BLANK;
+
+CPSubscriptionType cpSubscriptionType = cpDefinitionSubscriptionInfoDisplayContext.getCPSubscriptionType(subscriptionType);
+
+if (cpSubscriptionType != null) {
+	defaultCPSubscriptionTypeLabel = cpSubscriptionType.getLabel(locale);
+}
+
+CPSubscriptionTypeJSPContributor cpSubscriptionTypeJSPContributor = cpDefinitionSubscriptionInfoDisplayContext.getCPSubscriptionTypeJSPContributor(subscriptionType);
+
+boolean ending = maxSubscriptionCycles > 0;
```

Comment:
ugh, looks like this file may have just been moved... but git shows it as deleted + added. Oh well.

Don't set a boolean to the result of an operation like this. use `ending = false; if (blah) ending = true;` (with better formatting obviously)